lib/core/network/repository/auth_repository.dart

```diff
@@ -103,15 +118,18 @@ class AuthRepository {
         );
 
     if (shouldLoginAfter) {
-      loginUser();
+      loginUser(userProfileData, userAuthData);
     }
   }
 
-  void loginUser() {
-    return _controller.add(AuthenticationStatus.authenticated);
+  void loginUser(
+    UserProfileData userProfileData,
+    UserAuthData userAuthData,
+  ) {
+    return _controller.add(Authenticated(userAuthData, userProfileData));
```

Comment:
Does this take into account that a valid user may not yet have a profile data?

Example a brand new user. 

Example 2: A user imported from Telos using the private key.

We need to make sure that everywhere we use profile data, it may be null.